contrib/plugins/qt-creator-project.py

```diff
@@ -0,0 +1,231 @@
+        for root, directories, filenames in os.walk(newPath):
+            for filename in filenames:
+                if source.match(filename):
+                    sList.append(os.path.join(os.getcwd(), os.path.join(root,filename)))
+            # it's more faster to add all directories to includes and not to use regex, sort & remove duplicate entries
+            # this also helps qt-creator to resolve includes like <linux/bitops.h> which
+            # is not found in case there is no header in 'linux'
+            hList.append(os.path.join(os.getcwd(),root))
+
+    # get system default project id
+    # storred in ~/.config/QtProject/qtcreator/profiles.xml
+    # <value type="QString" key="PE.Profile.Id">{4bf97dcc-3970-4b8f-97cc-b4ce38f2b2d7}</value>
+    projectid = -1
+    with open(os.path.join(expanduser('~'), ".config/QtProject/qtcreator/profiles.xml")) as profilesFile:
+        for line in profilesFile:
+            if 'PE.Profile.Id' in line:
+                m = re.search('.*PE.Profile.Id">(.+?)</value>', line)
+                if m:
+                    projectid = m.group(1)
+    if projectid == -1:
+       print("No profile file found. Generated projects may not work")
+
+    # Generate the includes file
+    includesFile = os.path.join(destination, projectDir + ".includes")
+    generateFile(hList, includesFile)
+    # Generate files file
+    filesFile = os.path.join(destination, projectDir  + ".files")
+    generateFile(sList, filesFile)
+
+    if not updateOnly:
+        #Generate creator file
+        creatorFile = os.path.join(destination, projectDir  + ".creator")
+        generateFile([], creatorFile)
+        #Generate the creator.shared file using a template and modify some settings
+        sharedFile = os.path.join(destination, projectDir  + ".creator.shared")
+        templateFile = os.path.join(os.getcwd() + "/plugins/qt-project.template")
+        with open(sharedFile, 'w') as sharedFile:
+            with open(templateFile) as template:
+                for line in template:
+                    if 'plugin:id' in line:
+                        sharedFile.write(line.replace("<!-- plugin:id -->", projectid ))
+                    elif 'plugin:buildstep' in line:
+                        addBuildSteps(sharedFile, 'dev' if devMode else 'build', project, destination)
+                    else:
+                        sharedFile.write(line)
+
+def projectGenerator(argv, bobRoot):
+    parser = argparse.ArgumentParser(prog="bob project", description='Generate Project Files')
+    parser.add_argument('package', type=str, nargs='?',
+                        help="Sub-package to start listing from")
+    devBuild = parser.add_mutually_exclusive_group()
+    devBuild.add_argument('-d', '--dev', default=True, action='store_true',
+                        help="Use bob dev mode in project files")
+    devBuild.add_argument('-b', '--build', default=False, action='store_false', dest='dev',
```

Comment:
Release mode makes no sense for IDE projects. If the recipes are updated the location of the files and results might change. I don't think this makes sense in an IDE and will be quite surprising.